manager/controllers/baseline_update.go

```diff
@@ -0,0 +1,117 @@
+package controllers
+
+import (
+	"app/base"
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"app/manager/middlewares"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+)
+
+type UpdateBaselineRequst struct {
+	Name         string         `json:"name"`
+	InventoryIDs []SystemID     `json:"inventory_ids"`
+	Config       BaselineConfig `json:"config"`
+}
+
+type BaselineConfig struct {
+	ToTime string `json:"to_time"`
+}
+
+type AccountID struct {
+	RhAccountID int
+}
+type UpdateBaselineResponse struct {
+	BaselineID int
+}
+
+// @Summary Update a baseline for my set of systems
+// @Description Update a baseline for my set of systems
+// @ID updateBaseline
+// @Security RhIdentity
+// @Accept   json
+// @Produce  json
+// @Param    baseline_id path    int                  true "Baseline ID"
+// @Param    body        body    UpdateBaselineRequst true "Request body"
+// @Success 200 {object} UpdateBaselineResponse
+// @Router /api/patch/v1/baselines/{baseline_id} [post]
+func BaselineUpdateHandler(c *gin.Context) {
+	account := AccountID{
+		RhAccountID: c.GetInt(middlewares.KeyAccount),
+	}
+
+	var req UpdateBaselineRequst
+	if err := c.ShouldBindJSON(&req); err != nil {
+		LogAndRespBadRequest(c, err, "Invalid request body")
+	}
+
+	baselineID := c.Param("baseline_id")
+	if baselineID == "" {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "baseline_id param not found"})
+		return
+	}
+
+	var exists int64
+	err := database.Db.Model(&models.Baseline{}).
+		Where("id = ? ", baselineID).Count(&exists).Error
+	if err != nil {
+		LogAndRespError(c, err, "database error")
+	}
+	if exists == 0 {
+		LogAndRespNotFound(c, errors.New("Baseline not found"), "Baseline not found")
+		return
+	}
+
+	config, err := json.Marshal(&req.Config)
+	if err != nil {
+		LogAndRespError(c, err, "Invalid config")
+	}
+
+	ID, err := strconv.Atoi(baselineID)
+	if err != nil {
+		LogAndRespError(c, err, "Invalid baseline id")
+	}
+
+	baseline := models.Baseline{
+		ID:          ID,
+		Name:        req.Name,
+		Config:      config,
+		RhAccountID: account.RhAccountID,
+	}
+
+	err = buildUpdateBaselineQuery(baseline, req.InventoryIDs, account, c)
+	if err != nil {
+		LogAndRespError(c, err, "Database error")
+	}
+
+	c.JSON(http.StatusOK, baseline.ID)
+}
+
+//nolint: lll
+func buildUpdateBaselineQuery(baseline models.Baseline, inventoryIDs []SystemID, account AccountID, c *gin.Context) error {
+	tx := database.Db.WithContext(base.Context).Begin()
+	defer tx.Rollback()
+
+	if err := tx.Model(models.Baseline{}).Where("id = (?)", baseline.ID).Updates(&baseline).Error; err != nil {
```

Comment:
I think empty fields will not be updated this way. That's why we use update using map, so values are set explicitly. Check e.g. this: https://github.com/RedHatInsights/patchman-engine/blob/master/evaluator/evaluate.go#L293

Reply:
I believe it is ok (I may be wrong :) I want to just ask my thought), as the baseline provided from function call is already initialized manully here https://github.com/RedHatInsights/patchman-engine/pull/877/files#diff-c4f162b041d7ba3acd6dfb59cff7475a1b8b996e0929acdb58fefe2f548ccdeeR81. I think I would repeat myself one more time if create a map inside buildUpdateBaselineQuery function as well. I have checked updating endpoint is working as expected.

Reply:
Try to write test updating baseline Name from "temporary_baseline" to "". I think it won't work:
~~~golang
func TestUpdateBaseline2(t *testing.T) {
	utils.SkipWithoutDB(t)

	core.SetupTestEnvironment()
	baselineID := database.CreateBaseline(t)

	body := UpdateBaselineRequst{
		Name: "",
		InventoryIDs: []SystemID{"00000000-0000-0000-0000-000000000005", "00000000-0000-0000-0000-000000000006"},
	}

	bodyJSON, err := json.Marshal(&body)
	if err != nil {
		panic(err)
	}

	w := httptest.NewRecorder()
	path := fmt.Sprintf(`/%v`, baselineID)
	req, _ := http.NewRequest("POST", path, bytes.NewBuffer(bodyJSON))
	core.InitRouterWithParams(BaselineUpdateHandler, 1, "POST", "/:baseline_id").ServeHTTP(w, req)

	assert.Equal(t, http.StatusOK, w.Code)

	var baseline models.Baseline
	assert.Nil(t, database.Db.Model(&baseline).Where("id = ?", baselineID).Find(&baseline).Error)
	assert.Equal(t, baseline.ID, baselineID)
	assert.Equal(t, baseline.Name, "")
}
~~~

Reply:
Yes, it really did not work for empty fields. So I have created a map of baseline name and config. As these columns only will needs update. I have added one more test for empty fields.